tests/playwright/src/specs/docker-compatibility-smoke.spec.ts

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { execSync } from 'node:child_process';
+
+import type { Page } from '@playwright/test';
+
+import { DockerCompatibilityPage } from '../model/pages/docker-compatibility-page';
+import { PodmanMachineDetails } from '../model/pages/podman-machine-details-page';
+import { PreferencesPage } from '../model/pages/preferences-page';
+import { ResourcesPage } from '../model/pages/resources-page';
+import { SettingsBar } from '../model/pages/settings-bar';
+import { expect as playExpect, test } from '../utility/fixtures';
+import { isWindows } from '../utility/platform';
+import { waitForPodmanMachineStartup } from '../utility/wait';
+
+const defaultMachine = 'Podman Machine';
+const DCPreferencesHref = '/preferences/default/preferences.dockerCompatibility';
+const DCSettingsHref = '/preferences/docker-compatibility';
+
+test.beforeAll(async ({ runner, welcomePage, page }) => {
+  runner.setVideoAndTraceName('docker-compatibility-e2e');
+
+  await welcomePage.handleWelcomePage(true);
+  await waitForPodmanMachineStartup(page);
+});
+
+test.afterAll(async ({ runner, page }) => {
+  if (test.info().status === 'failed') {
+    await setDockerCompatibilityFeature(false, page);
+
+    try {
+      // eslint-disable-next-line sonarjs/no-os-command-from-path
+      execSync('podman machine start');
+      console.log('Default podman machine started');
+    } catch (error) {
+      if (error instanceof Error && error.message.includes('VM already running')) {
+        console.log('Default podman machine already started, skipping start.');
+      }
+    }
+  }
+  await runner.close();
+});
+
+//Feature unstable on mac and linux atm
+test.skip(!isWindows, 'Testing only on Windows');
+
+test.describe.serial('Verify docker compatibility feature', { tag: '@smoke' }, () => {
+  test('Enable the docker compatibility experimental feature', async ({ navigationBar, page }) => {
+    //Verify Settings
+    await navigationBar.openSettings();
+    const settingsBar = new SettingsBar(page);
+
+    const DCLink = settingsBar.getLinkLocatorByHref(DCSettingsHref);
+    await playExpect(DCLink).not.toBeVisible();
+
+    //Enable the feature
+    await setDockerCompatibilityFeature(true, page);
+  });
+  test('Verify Docker Compatibility page', async ({ page }) => {
+    const settingsBar = new SettingsBar(page);
+    const dockerCompatibilityPage = await settingsBar.openTabPage(DockerCompatibilityPage);
+    await playExpect(dockerCompatibilityPage.heading).toBeVisible();
+    playExpect(await dockerCompatibilityPage.socketIsReachable()).toBeTruthy();
```

Comment:
I would again replace simple check for polling mechanism to make sure the state had time to be rendered.